### PR TITLE
Priority queue questions

### LINKED ISSUE #853: [Feature Request] Adding sums in Priority queue
### Is your feature request related to a problem? Please describe.
I'm frustrated when I need to frequently compute the sum of elements in a priority queue. Currently, I have to iterate through all elements, which is inefficient, especially with large datasets.

### Describe the solution you'd like
I would like to add a method to the priority queue implementation that calculates and returns the sum of all elements in the queue. This method should operate in constant time \(O(1)\) by maintaining a running sum as elements are added or removed from the queue.

### Describe alternatives you've considered
As an alternative, I considered implementing a separate data structure to keep track of the sums. However, this would introduce additional complexity and overhead, and I believe enhancing the existing priority queue is a more efficient solution.

### Additional context
Maintaining a running sum in the priority queue would greatly enhance its usability for applications that frequently require both priority operations and sum calculations. This feature could be particularly useful in scenarios such as real-time data processing and financial calculations.

---

**Would you like to work on this feature?**  
- [x] Yes  
- [ ] No

### PR DESCRIPTION
**Pull request by monishkumardvs**
### Description
added questions  about priority queue
image

Fixes #853

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

### Checklist:

- [x ] My code follows the style guidelines of this project
- [x ] I have performed a self-review of my code
- [ x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by monishkumardvs** [comment]
Can u check this pr and give the labels

File: docs/basic-dsa/Strings/Advanced-techniques.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
+---
+id: Advanced String
```
Comment (ajay-dhangar) [inline docs/basic-dsa/Strings/Advanced-techniques.md]: replace `id: Advanced String` with `id: advanced-string` for best practice

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
Update your code based on suggestions and for more details, you can check out our CONTRIBUTING.md file.

File: docs/basic-dsa/Strings/Advanced-techniques.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
+---
+id: advanced-string
```
Comment (monishkumardvs) [inline docs/basic-dsa/Strings/Advanced-techniques.md:2]: Added the changes in advanced strings

File: docs/basic-dsa/Strings/Advanced-techniques.md (hunk 1 above)
Comment (monishkumardvs) [reply docs/basic-dsa/Strings/Advanced-techniques.md]: i changed this

**Comment by monishkumardvs** [comment]
Can u tell wht i shld change

**Review by ajay-dhangar — APPROVED** [review]
Done

**Pull request merged**